### PR TITLE
Fix undeterministic response tests 2

### PR DESCRIPTION
**Pull request by armingeiser**

### CODE REVIEW

File: applications/StructuralMechanicsApplication/tests/response_function_tests/rectangular_plate_adjoint_parameters.json [hunk 1]
```diff
@@ -85,7 +85,7 @@
             "node_output"         : false,
             "skin_output"         : false,
             "plane_output"        : [],
-            "nodal_results"       : ["DISPLACEMENT", "SHAPE_SENSITIVITY"],
+            "nodal_results"       : ["ADJOINT_DISPLACEMENT", "ADJOINT_ROTATION", "DISPLACEMENT", "SHAPE_SENSITIVITY"],
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/response_function_tests/rectangular_plate_adjoint_parameters.json:88]: why don't you deactivate the output by default?

**Comment by philbucher** [comment]
Some tests are still failing, not sure abt the reason, might be numerical
The `TestAdjointStressResponseFunction` fails only sometimes, the `TestMassResponseFunction` always
~~~
======================================================================
FAIL: test_execution (__main__.TestAdjointStressResponseFunction)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "structural_response_function_test_factory.py", line 130, in test_execution
    self.assertAlmostEqual(self.gradient[4][2], 1.1013492791909624e-05, 9)
AssertionError: 1.1012946954476409e-05 != 1.1013492791909624e-05 within 9 places

======================================================================
FAIL: test_execution (__main__.TestMassResponseFunction)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "structural_response_function_test_factory.py", line 142, in test_execution
    self.assertAlmostEqual(self.gradient[nodeId][1], 2.2737367544323206e-07, 9)
AssertionError: 4.547473508864641e-07 != 2.2737367544323206e-07 within 9 places

----------------------------------------------------------------------
~~~

File: applications/StructuralMechanicsApplication/tests/response_function_tests/rectangular_plate_adjoint_parameters.json (hunk 1 above)
Comment (armingeiser) [reply applications/StructuralMechanicsApplication/tests/response_function_tests/rectangular_plate_adjoint_parameters.json:88]: :+1:

**Comment by philbucher** [comment]
The `TestAdjointStrainEnergyResponseFunction` still fails in every second run:
~~~
======================================================================
FAIL: test_execution (__main__.TestAdjointStrainEnergyResponseFunction)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "structural_response_function_test_factory.py", line 93, in test_execution
    self.assertAlmostEqual(self.gradient[4][2], -1.22992394412451e-05, 10)
AssertionError: -1.229918470505571e-05 != -1.22992394412451e-05 within 10 places

----------------------------------------------------------------------
~~~

**Review by philbucher — APPROVED** [review]
now it works :+1:

**Pull request merged**